server/src/task/test/task.e2e-spec.ts

```diff
@@ -0,0 +1,170 @@
+        .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+        .set('x-csrf-token', csrfToken)
+        .expect(201);
+    });
+  });
+
+  // get-task test
+
+  describe('GET /tasks/:taskId', () => {
+    const taskId = 126;
+    it('should return 401 if not authenticated', async () => {
+      await request(app.getHttpServer()).get(`/tasks/${taskId}`).expect(401);
+    });
+
+    it('should return 200 and the task data if authenticated', async () => {
+      return request(app.getHttpServer())
+        .get(`/tasks/${taskId}`)
+        .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+        .set('x-csrf-token', csrfToken)
+        .expect(200);
+    });
+  });
+
+  // update-task test
+
+  describe('PATCH /tasks/:taskId', () => {
+    const taskId = 126;
+    it('should return 401 if not authenticated', async () => {
+      await request(app.getHttpServer()).patch(`/tasks/${taskId}`).expect(401);
+    });
+
+    // invalid test
+    invalidUpdateTaskFields.forEach((invalidTaskField, index) => {
+      it(`should return 400 for invalid task fields - case ${index + 1}`, async () => {
+        await request(app.getHttpServer())
+          .patch(`/tasks/${taskId}`)
+          .send(invalidTaskField)
+          .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+          .set('x-csrf-token', csrfToken)
+          .expect(400);
+      });
+    });
+
+    // valid test
+    it('should return 200 and task data if authenticated', async () => {
+      await request(app.getHttpServer())
+        .patch(`/tasks/${taskId}`)
+        .send(validUpdateTaskField)
+        .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+        .set('x-csrf-token', csrfToken)
+        .expect(200);
+    });
+  });
+
+  // update-task-status test
+
+  describe('PATCH /tasks/:taskId/status', () => {
+    it('should return 401 if not authenticated', async () => {
+      await request(app.getHttpServer()).patch('/tasks/126/status').expect(401);
+    });
+
+    const updateTaskStatus = async (taskId: number, isCompleted: boolean) => {
+      return request(app.getHttpServer())
+        .patch(`/tasks/${taskId}/status`)
+        .send({ isCompleted })
+        .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+        .set('x-csrf-token', csrfToken)
+        .expect(200);
+    };
+
+    // set status to true
+    it('should return 200 and task status if authenticated', async () => {
+      const taskId = 126;
+      const response = await updateTaskStatus(taskId, true);
+      expect(response.body).toMatchObject({ isCompleted: true });
+    });
+
+    // set status to false
+    it('should return 200 and task status if authenticated', async () => {
+      const taskId = 127;
+      const response = await updateTaskStatus(taskId, false);
+      expect(response.body).toMatchObject({ isCompleted: false });
+    });
+
+    // delete-task test
+
+    describe('DELETE /tasks/:taskId', () => {
```

Comment:
Delete should be outside of `PATCH /tasks/:taskId/status` test!

Reply:
Fixed it!
https://github.com/HAK2024/fairy-share/pull/69/commits/4ebfcf216f2c0a5b93c66365c6b9419996cfcc3b